Gems/ROS2/Code/Source/RobotImporter/SDFormat/Hooks/ROS2SkidSteeringModelHook.cpp

```diff
@@ -0,0 +1,257 @@
+                    wheelPairs,
+                    configuration.m_axles.size());
+            }
+
+            AZ_Warning(
+                "ROS2SkidSteeringModelPluginHook",
+                !configuration.m_axles.empty(),
+                "VehicleConfiguration parsing error: cannot find any axles.");
+
+            return configuration;
+        }
+
+        VehicleDynamics::SkidSteeringModelLimits GetModelLimits(const sdf::ElementPtr element)
+        {
+            VehicleDynamics::SkidSteeringModelLimits modelLimits;
+            if (element->HasElement("wheelAcceleration"))
+            {
+                modelLimits.SetAngularAccelerationLimit(element->Get<float>("wheelAcceleration"));
+            }
+            else if (element->HasElement("max_wheel_acceleration"))
+            {
+                modelLimits.SetAngularAccelerationLimit(element->Get<float>("max_wheel_acceleration"));
+            }
+            else
+            {
+                AZ_Warning("ROS2SkidSteeringModelPluginHook", false, "VehicleConfiguration parsing error: cannot determine model limits.");
+            }
+
+            return modelLimits;
+        }
+    } // namespace Parser
+
+    ModelPluginImporterHook ROS2ModelPluginHooks::ROS2SkidSteeringModel()
+    {
+        ModelPluginImporterHook importerHook;
+        importerHook.m_pluginNames =
+            AZStd::unordered_set<AZStd::string>{ "libgazebo_ros_skid_steer_drive.so", "libgazebo_ros_diff_drive.so" };
+
+        importerHook.m_sdfPluginToComponentCallback =
+            [](AZ::Entity& entity, const sdf::Plugin& sdfPlugin, const sdf::Model& sdfModel, const CreatedEntitiesMap& createdEntities)
+            -> ModelPluginImporterHook::ConvertPluginOutcome
+        {
+            // Parse parameters
+            const sdf::ElementPtr element = sdfPlugin.Element();
+            VehicleDynamics::VehicleConfiguration vehicleConfiguration = Parser::GetConfiguration(element, sdfModel, createdEntities);
+            VehicleDynamics::SkidSteeringModelLimits modelLimits = Parser::GetModelLimits(element);
+
+            // Create required components
+            HooksUtils::CreateComponent<ROS2FrameComponent>(entity);
+            HooksUtils::CreateComponent<ROS2RobotControlComponent>(entity);
+            HooksUtils::CreateComponent<VehicleDynamics::SkidSteeringModelComponent>(
+                entity, vehicleConfiguration, VehicleDynamics::SkidSteeringDriveModel(modelLimits));
+
+            // Create Skid Steering Control Component
+            if (HooksUtils::CreateComponent<SkidSteeringControlComponent>(entity))
+            {
+                return AZ::Success();
+            }
+            else
+            {
+                return AZ::Failure(AZStd::string("Failed to create ROS2 Skid Steering Control Component"));
+            }
+        };
+
+        return importerHook;
+    }
+} // namespace ROS2::SDFormat
```

Comment:
Note: good to think what to abstract here as ackermann drive support will be added to avoid any copying.